packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -39,10 +63,34 @@ class RenderParagraph extends RenderBox {
     markNeedsPaint();
   }
 
+  /// Whether the text should break at soft line breaks.
+  ///
+  /// If false, the glyphs in the text will be positioned as if there was unlimited horizontal space.
+  bool get softWrap => _softWrap;
+  bool _softWrap;
+  void set softWrap(bool value) {
+    assert(value != null);
+    if (_softWrap == value)
+      return;
+    _softWrap = value;
+    markNeedsLayout();
+  }
+
+  /// How visual overflow should be handled.
+  TextOverflow get overflow => _overflow;
+  TextOverflow _overflow;
+  void set overflow(TextOverflow value) {
+    assert(value != null);
+    if (_overflow == value)
+      return;
+    _overflow = value;
+    markNeedsPaint();
+  }
+
   void _layoutText(BoxConstraints constraints) {
     assert(constraints != null);
     assert(constraints.debugAssertIsValid());
-    _textPainter.layout(minWidth: constraints.minWidth, maxWidth: constraints.maxWidth);
+    _textPainter.layout(minWidth: constraints.minWidth, maxWidth: _softWrap ? constraints.maxWidth : double.INFINITY);
```

Comment:
In principle it should work because this is just the max width.  We layout the text twice: once to get the intrinsic width and once at that width (subject to these constraints).

With libtxt, we can do this more efficiently because we'll be able to compute the intrinsic sizes without actually doing a fully layout.